### ISSUE TITLE
Discriminator value change ignored in findOneAndUpdate

### ISSUE THREAD

**Issue by ethanresnick** [Automattic/mongoose#5613]
**Do you want to request a *feature* or report a *bug*?**

Report a bug, I believe.

**What is the current behavior?**

```js
const Organization = mongoose.model("Organization", OrgSchema);
const School =  Organization.discriminator("School", SchoolSchema);

// The update below runs successfully, 
// but the discriminator key in the update doc is ignored
Organization.findByIdAndUpdate(someOrgId, { 
  [OrgSchema.options.discriminatorKey]: 'School' 
}).exec()
```

**What is the expected behavior?**
The update should change the value of the discriminator field on the matching document.

**Please mention your node.js, mongoose and MongoDB version.**
Node 8.2.1, mongoose 4.11.9, mongo 3.4.6

**Comment by vkarpov15** [comment]
As a workaround, add the discriminator key as a path in the base schema:

```javascript
const mongoose = require('mongoose');

mongoose.connect('mongodb://localhost:27017/test');
mongoose.Promise = Promise;
mongoose.set('debug', true);

const OrgSchema = new mongoose.Schema({ name: String, __t: String });
const SchoolSchema = new mongoose.Schema({ principal: String });
const Organization = mongoose.model("Organization", OrgSchema);
const School =  Organization.discriminator("School", SchoolSchema);

// The update below runs successfully, 
// but the discriminator key in the update doc is ignored
Organization.findOneAndUpdate({}, {         
  [OrgSchema.options.discriminatorKey]: 'School' 
}).exec()
```

**Comment by vkarpov15** [comment]
Fix will be in 4.12. The solution is to always add the discriminator key to the base schema, which is a bigger change than I'm comfortable releasing in a patch release. We'll release this with 4.12.0.

**Comment by ethanresnick** [comment]
@vkarpov15 Sounds good. Thanks for your responsiveness here and the temporary workaround.
